### PR TITLE
complete task (unit testing frameworks 12673)

### PR DESCRIPTION
**Pull request by Irina-N23**
Task #12673 has been completed.
Just check whether everything is correct in it!

Here is a link to the JIRA issue with a screenshot:
https://jira.epam.com/jira/browse/EPMFARMATS-12673

### CODE REVIEW

File: unit-testing-frameworks/app/calculatorEntity.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+"use strict";
+
+const utilities = require("./utilities");
```
Comment (SiarheiRylach) [inline unit-testing-frameworks/app/calculatorEntity.js]: sugar: `{isArrayComposedOfNumbers} = require("./utilities");`

File: unit-testing-frameworks/app/utilities.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    const isNumbers = array.every(element => typeof element === "number");
+
+    if (isNumbers) {
+        return true;
+    } else {
+        throw new TypeError("(!) Incorrect type of parameters: the type must be \"number\".\n");
+    }
+}
+
+module.exports.isArrayComposedOfNumbers = isArrayComposedOfNumbers;
```
Comment (SiarheiRylach) [inline unit-testing-frameworks/app/utilities.js]: sugar: 
```
module.exports.isArrayComposedOfNumbers = function (array) {
    if (array.length === 0) {
        throw new Error("(!) No arguments have been inputted.\n")
    }

    const isNumbers = array.every(element => typeof element === "number");

    if (isNumbers) {
        return true;
    } else {
        throw new TypeError("(!) Incorrect type of parameters: the type must be \"number\".\n");
    }
};
```

File: unit-testing-frameworks/test/specs/add.spec.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    beforeEach(() => calculator = new CalculatorEntity());
+    afterEach(() => calculator = null);
+
+    it("must return sum of inputted arguments if they are integers",
+        () => expect(calculator.add(-43, 11, 35)).to.be.equal(3));
+
+    it("must return sum of inputted arguments if they include integers"
+                                             +" and floating-point numbers",
+        () => expect(calculator.add(5.5, 11, 15.2, -1.3)).to.be.equal(30.4));
```
Comment (SiarheiRylach) [inline unit-testing-frameworks/test/specs/add.spec.js:17]: you don't need to verify how JS works.

File: unit-testing-frameworks/test/specs/add.spec.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+
+describe("method \"add()\": negative scenarios", function() {
+    let calculator;
+
+    beforeEach(() => calculator = new CalculatorEntity());
+    afterEach(() => calculator = null);
+
+    it("must throw an error if arguments have not been inputted",
+        () => expect(calculator.add.bind())
```
Comment (SiarheiRylach) [inline unit-testing-frameworks/test/specs/add.spec.js]: why bind?

File: unit-testing-frameworks/app/calculatorEntity.js (hunk 1 above)
Comment (Irina-N23) [reply unit-testing-frameworks/app/calculatorEntity.js]: ok, done!

File: unit-testing-frameworks/app/utilities.js (hunk 2 above)
Comment (Irina-N23) [reply unit-testing-frameworks/app/utilities.js]: ok, done!

File: unit-testing-frameworks/test/specs/add.spec.js (hunk 3 above)
Comment (Irina-N23) [reply unit-testing-frameworks/test/specs/add.spec.js:17]: Thank you! Ok, I'll take that into account for my future tasks.

File: unit-testing-frameworks/test/specs/add.spec.js (hunk 4 above)
Comment (Irina-N23) [reply unit-testing-frameworks/test/specs/add.spec.js]: another way of fulfilling that assertion has been also applied

**Review by SiarheiRylach — APPROVED** [review]

**Pull request merged**